src/contracts/core/RewardsCoordinator.sol

```diff
@@ -340,6 +349,83 @@ contract RewardsCoordinator is
         }
     }
 
+    /**
+     * @notice Validate a PerformanceRewardsSubmission. Called from `createAVSPerformanceRewardsSubmission`.
+     * @param performanceRewardsSubmission PerformanceRewardsSubmission to validate.
+     * @return total amount to be transferred from the avs to the contract.
+     */
+    function _validatePerformanceRewardsSubmission(
```

Comment:
Thoughts on using `revert` with custom errors? Or just custom errors in general instead of error strings? I kept it similar to the current setup but using custom errors will be more gas efficient.

Reply:
let's do that with slashing like we are on #727

Reply:
Alright holding off on custom errors for now.